src/gluonts/dataset/tabular.py

```diff
@@ -0,0 +1,269 @@
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+from dataclasses import dataclass, field
+from datetime import datetime
+from typing import Iterator, List, Union, Dict
+
+import pandas as pd
+
+from gluonts.dataset.common import Dataset, DataEntry, ProcessDataEntry
+from gluonts.dataset.field_names import FieldName
+
+
+@dataclass
+class DataFramesDataset(Dataset):
+    """
+    A pd.DataFrame-based dataset type.
+
+    This class is constructed with a collection of pd.DataFrame-objects where each
+    DataFrame is representing one time series. A target and a timestamp columns are
+    essential. Furthermore, static/dynamic real/categorical features can be specified.
+
+    Parameters
+    ----------
+    dataframes: Union[List[pd.DataFrame], Dict[str, pd.DataFrame]]
+        List or Dict of 'pd.DataFrame's containing at least timestamp and target columns.
+        If a Dict is provided, the key will be the associated 'item_id'.
+    target: str or List[str]
+        Name of the column that contains the target time series.
+        For multivariate targets `target` is expecting a list of column names.
+    timestamp: str
+        Name of the column that contains the timestamp information.
+    freq: str
+        Frequency of observations in the time series. Must be a valid pandas frequency.
+    feat_dynamic_real: List[str]
+        List of column names that contain dynamic real features.
+    feat_dynamic_cat: List[str]
+        List of column names that contain dynamic categorical features.
+    feat_static_real: List[str]
+        List of column names that contain static real features.
+    feat_static_cat: List[str]
+        List of column names that contain static categorical features.
+    past_feat_dynamic_real: List[str]
+        List of column names that contain dynamic real features only for the history.
+    prediction_length: int
+        For target and past dynamic features last `prediction_length` elements are
+        removed when iterating over the data set. This becomes important when the
+        predictor is called.
+    """
+
+    dataframes: Union[List[pd.DataFrame], Dict[str, pd.DataFrame]]
+    target: Union[str, List[str]]
+    timestamp: str
+    freq: str
+    feat_dynamic_real: List[str] = field(default_factory=list)
+    feat_dynamic_cat: List[str] = field(default_factory=list)
+    feat_static_real: List[str] = field(default_factory=list)
+    feat_static_cat: List[str] = field(default_factory=list)
+    past_feat_dynamic_real: List[str] = field(default_factory=list)
+    prediction_length: int = 0
+
+    def __post_init__(self) -> None:
+        self.one_dim_target = not isinstance(self.target, list)
+        self.process = ProcessDataEntry(
+            self.freq, one_dim_target=self.one_dim_target
+        )
+        if isinstance(self.dataframes, dict):
```

Comment:
Can you add a comment how we store dataframes? What is the key of the dict used for?